### DISCUSSION TITLE
My amd64 images are being overwritten by my arm64 images?

### DISCUSSION THREAD

**Discussion by Ilyes512** [docker/build-push-action#1073] [Q&A]
I have https://github.com/Ilyes512/php83-fpm repository that builds and pushes images to the Github registry using this action. I do that in 2 different jobs, one for each of the architectures: amd64 and arm64. 

When the jobs run the amd64 job it is way quicker because of building in its native architecture. And so when I do a manifest inspect after the amd64 job is done I would see something like this:

```
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.oci.image.index.v1+json",
   "manifests": [
      {
         "mediaType": "application/vnd.oci.image.manifest.v1+json",
         "size": 2759,
         "digest": "sha256:9e7acd536ecb303743049c1d0ba69f44448bdfdabcef716b905334486cfead30",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.oci.image.manifest.v1+json",
         "size": 567,
         "digest": "sha256:a2ba87147b42dfb219aa9cd787ee316ed7bc2d07569a22bd8de2d9ec6aa2426f",
         "platform": {
            "architecture": "unknown",
            "os": "unknown"
         }
      }
   ]
}
```

But as soon as the arm64 version is done it seems to replace the amd64 version. A manifest inspect would show:

```
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.oci.image.index.v1+json",
   "manifests": [
      {
         "mediaType": "application/vnd.oci.image.manifest.v1+json",
         "size": 2759,
         "digest": "sha256:781b0f0d78366c102c9d90a8683533265aebce1060bce95aa6a40cb51274198a",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.oci.image.manifest.v1+json",
         "size": 567,
         "digest": "sha256:683e2bfb5e644979593809a83b5402661dbda165cc87f4e2c939b9ff8427e20d",
         "platform": {
            "architecture": "unknown",
            "os": "unknown"
         }
      }
   ]
}
```

Why does this happen? Do I need to push both architecture images at the same time? I thought it would "append" the architecture to the manifest.

**Comment by crazy-max** [comment]
That makes sense as you have two dedicated builds so one build takes over tags you're pushing. You need to use the `platforms` input to build a multi-platform image so manifests are merged when both are built. You can also distribute builds across runners. More info: https://docs.docker.com/build/ci/github-actions/multi-platform/
